### PR TITLE
Move value parsing into its own function

### PR DESCRIPTION
**Pull request by Schamper**

### CODE REVIEW

File: dissect/regf/regf.py [hunk 1]
```diff
@@ -532,6 +486,51 @@ def try_decode_sz(data):
...
+        data_len = len(data)
+        data = BytesIO(data)
+
+        multi_string = []
+        while data.tell() < data_len:
+            string = read_null_terminated_wstring(data)
+            if string == "":
+                break
+
+            multi_string.append(string)
```
[the remark this answers is not in the document]
Comment (pyrco) [reply dissect/regf/regf.py:515]: Wouldn't that just error at the end of the data?

File: dissect/regf/regf.py [hunk 2]
```diff
@@ -532,6 +486,51 @@ def try_decode_sz(data):
         return data.decode("utf-16-le", "ignore").strip("\x00")
 
 
+def parse_value(data_type, data):
```
[the remark this answers is not in the document]
Comment (pyrco) [reply dissect/regf/regf.py:489]: I think it is `Union[int, str, list[str], bytes]`

File: dissect/regf/regf.py [hunk 3]
```diff
@@ -532,6 +486,51 @@ def try_decode_sz(data):
...
+    if data_type == REG_QWORD:
+        return c_regf.uint64(data) if len(data) else 0
+
+    if data_type == REG_FULL_RESOURCE_DESCRIPTOR:
+        log.warning("Unimplemented REG_FULL_RESOURCE_DESCRIPTOR")
+        return data
+
+    if data_type == REG_RESOURCE_REQUIREMENTS_LIST:
+        log.warning("Unimplemented REG_RESOURCE_REQUIREMENTS_LIST")
+        return data
```
[the remark this answers is not in the document]
Comment (Schamper) [reply dissect/regf/regf.py:528]: I don't intend to make such large changes with this PR. This PR is only for moving the location of the function.

File: dissect/regf/regf.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Schamper) [reply dissect/regf/regf.py:519]: I also don't intend to make much changes to how the function works.

**Pull request merged**